x-pack/legacy/plugins/code/public/components/codeblock/embedded_codeblock.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiButtonIcon, EuiFlexGroup, EuiLink, EuiText, EuiTextColor } from '@elastic/eui';
+
+import { IPosition } from 'monaco-editor';
+import { CodeBlock } from './codeblock';
+import { Snippet } from '../integrations/data';
+
+const noop = () => {};
+
+export interface Props {
+  fileName: string;
+  header?: (externalLink: React.ReactNode) => React.ReactNode;
+  lineNumber: number | string;
+  onCodeClick?: (event: IPosition) => void;
+  snippet: Snippet;
+}
+
+export const EmbeddedCodeBlock = ({
+  fileName,
+  header: headerProp,
+  lineNumber,
+  onCodeClick,
+  snippet,
+}: Props) => {
+  const { compositeContent, language } = snippet;
+  const { content, lineMapping, ranges } = compositeContent;
+  const fileUrl = `#/${snippet.uri}/blob/HEAD/${snippet.filePath}`;
+
+  const externalLink = (
+    <EuiButtonIcon
+      className="integrations__link--external integrations__button-icon"
+      iconType="codeApp"
+      href={fileUrl}
+    />
+  );
+
+  const header = headerProp ? (
+    headerProp(externalLink)
+  ) : (
+    <EuiFlexGroup
+      className="integrations__snippet-info"
+      alignItems="center"
+      justifyContent="spaceBetween"
+      gutterSize="none"
+    >
+      <EuiText size="s">
+        <EuiLink href={fileUrl}>{fileName}</EuiLink>
+        <span> at </span>
+        <EuiLink href={fileUrl}>line {lineNumber}</EuiLink>
+      </EuiText>
+      <EuiText size="xs">
```

Comment:
do we need apply a flag from the props as an option to hide this one?

Reply:
obsolete comment, I believe.